script/20240516-revoke-roles/02_GrantAdminRoleForMultisig_Mainnet.s.sol

```diff
@@ -0,0 +1,362 @@
+import { RNSOperation } from "src/utils/RNSOperation.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { INSAuction } from "src/interfaces/INSAuction.sol";
+import { INSDomainPrice } from "src/interfaces/INSDomainPrice.sol";
+
+contract Migration__02_GrantAdminRoleForMultisig_Mainnet is Migration {
+  using Strings for *;
+  using ErrorHandler for bool;
+  using LibRNSDomain for string;
+
+  address duke = 0x0F68eDBE14C8f68481771016d7E2871d6a35DE11;
+  address multisig = 0x9d05d1f5b0424f8fde534bc196ffb6dd211d902a;
+
+  RNSUnified internal _rns;
+  RNSAuction internal _auction;
+  NameChecker internal _nameChecker;
+  RNSDomainPrice internal _domainPrice;
+  PublicResolver internal _publicResolver;
+  RNSReverseRegistrar internal _reverseRegistrar;
+  RONRegistrarController internal _ronController;
+  OwnedMulticaller internal _ownedMulticaller;
+  RNSOperation internal rnsOperation;
+  address internal _batchTransfer;
+
+  function run() external onlyOn(DefaultNetwork.RoninMainnet.key()) {
+    _rns = RNSUnified(loadContract(Contract.RNSUnified.key()));
+    _auction = RNSAuction(loadContract(Contract.RNSAuction.key()));
+    _nameChecker = NameChecker(loadContract(Contract.NameChecker.key()));
+    _domainPrice = RNSDomainPrice(loadContract(Contract.RNSDomainPrice.key()));
+    _publicResolver = PublicResolver(loadContract(Contract.PublicResolver.key()));
+    _reverseRegistrar = RNSReverseRegistrar(loadContract(Contract.RNSReverseRegistrar.key()));
+    _ronController = RONRegistrarController(loadContract(Contract.RONRegistrarController.key()));
+    // Verify: https://app.roninchain.com/address/0x27876429DB2cDDF017DBb63560D0366E4B4E6f8a
+    _ownedMulticaller = OwnedMulticaller(0x27876429DB2cDDF017DBb63560D0366E4B4E6f8a);
+    _batchTransfer = loadContract(Contract.ERC721BatchTransfer.key());
+    // Verify: https://app.roninchain.com/address/0xCD245263eDdEE593a5A66f93f74C58c544957339
+    rnsOperation = RNSOperation(0xCD245263eDdEE593a5A66f93f74C58c544957339);
+
+    address[] memory contracts = new address[](5);
+    contracts[0] = address(_domainPrice);
+    contracts[1] = address(_ronController);
+    contracts[2] = address(_nameChecker);
+    contracts[3] = address(_rns);
+    contracts[4] = address(_auction);
+
+    vm.startBroadcast(duke);
+    // Transfer .ron domain ownership to owned multicaller
+    uint256[] memory ids = new uint256[](3);
+    ids[0] = 0x0;
+    ids[1] = LibRNSDomain.RON_ID;
+    ids[2] = LibRNSDomain.ADDR_REVERSE_ID;
+    _rns.setApprovalForAll(_batchTransfer, true);
+
+    // Bulk transfer .ron domain ownership to owned multicaller
+    (bool success, bytes memory returnOrRevertData) = _batchTransfer.call(
+      abi.encodeWithSignature("safeBatchTransfer(address,uint256[],address)", _rns, ids, _ownedMulticaller)
+    );
+    success.handleRevert(returnOrRevertData);
+
+    // Remove approval for batch transfer
+    _rns.setApprovalForAll(_batchTransfer, false);
```

Comment:
if this tx is broadcasted out of order, the previous call might fail. Consider to revoke manually.